### PR TITLE
Initial implementation of analysis.R

### PR DESCRIPTION
**Pull request by magland**
This is a mashup of our latest main with #94 to implement the analysis.R window.

There is some duplicated code and some awkward imports, but it works.

For quick testing:

```R
plot(c(1, 2, 3, 4, 5), c(3, 7, 8, 9, 12))
print(2)
```

@WardBrian I'll hand it over to you to figure out the draws object next. If you need to start over again, feel free.

### CODE REVIEW

**Comment by WardBrian** [comment]
Great, thanks!

I'll try to get something going as a base later today or tomorrow (hopefully after #151 is merged so there aren't any conflict concerns if I do some renaming/reorganzing)

File: gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx [hunk 1]
```diff
@@ -96,6 +106,36 @@ const LeftPane: FunctionComponent<LeftPaneProps> = ({
...
+    ) : language === "r" ? (
+      <AnalysisRFileEditor
+        fileName="analysis.R"
+        fileContent={data.analysisPyFileContent}
+        editedFileContent={data.ephemera.analysisPyFileContent}
+        setEditedFileContent={(content) => {
+          update({
+            type: "editFile",
+            content,
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
```
Comment (jsoules) [inline gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx:118]: s.b. `ANALYSISRFILE`

File: gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx [hunk 2]
```diff
@@ -96,6 +106,36 @@ const LeftPane: FunctionComponent<LeftPaneProps> = ({
...
+          });
+        }}
+        onSaveContent={() => {
+          update({
+            type: "commitFile",
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
```
Comment (jsoules) [inline gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx:124]: s.b. `ANALYSISRFILE`

File: gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx [hunk 3]
```diff
@@ -96,6 +106,36 @@ const LeftPane: FunctionComponent<LeftPaneProps> = ({
         readOnly={false}
         spData={spData}
       />
+    ) : language === "r" ? (
+      <AnalysisRFileEditor
+        fileName="analysis.R"
```
Comment (jsoules) [inline gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx:111]: Should probably be a reference to the enum

File: gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx [hunk 4]
```diff
@@ -72,8 +81,9 @@ const LeftPane: FunctionComponent<LeftPaneProps> = ({
       return undefined;
     }
   }, [status, draws, numChains, paramNames]);
-  return (
-    <Splitter direction={SplitDirection.Vertical} initialSizes={[60, 40]}>
+
+  const editor =
+    language === "python" ? (
       <AnalysisPyFileEditor
         fileName="analysis.py"
```
Comment (jsoules) [inline gui/src/app/pages/HomePage/AnalysisWindow/AnalysisWindow.tsx:88]: This also should've been a reference to the enum

**Comment by magland** [comment]
Replaced by #157

**Pull request closed without merging**